sympy/sets/tests/test_fancysets.py

```diff
@@ -326,8 +327,8 @@ def test_Range_set():
     assert S(builtin_range(1000000000000)) == Range(1000000000000)
 
     # test Range.as_relational
-    assert Range(1, 4).as_relational(x) == (x >= 1) & (x <= 3) & Eq(x, floor(x))
-    assert Range(oo, 1, -2).as_relational(x) == (x >= 3) & (x < oo) & Eq(x, floor(x))
+    assert Range(1, 4).as_relational(x) == (x >= 1) & (x <= 3) & Eq(x, floor(x)) & Eq(Mod(x,1),0)
+    assert Range(oo, 1, -2).as_relational(x) == (x >= 3) & (x < oo) & Eq(x, floor(x)) & Eq(Mod(x,-2),1)
```

Comment:
Is this test even running? `md` as calculated above would be NaN.

Reply:
It looks like the problem is that the entire boolean evaluates to False on both sides.

Reply:
ahh. mod by negative is always nonpositive so it can't be 1